src/overlays/actors/ovl_En_Rz/z_en_rz.c

```diff
@@ -42,80 +39,647 @@ const ActorInit En_Rz_InitVars = {
+                this->actor.flags &= ~ACTOR_FLAG_10000;
+                func_800B8614(&this->actor, play, 120.0f);
+            }
+        }
+
+        if (!EN_RZ_GET_PARAM_8000(&this->actor)) {
+            sp30.x = this->actor.projectedPos.x;
+            sp30.y = this->actor.projectedPos.y;
+            sp30.z = this->actor.projectedPos.z;
+            func_801A1FB4(3, &sp30, 0x2A, 900.0f);
+        }
+    }
+}
+
+void func_80BFC608(EnRz* this, PlayState* play) {
+    EnRz_UpdateSkelAnime(this, play);
+
+    if ((Message_GetState(&play->msgCtx) == TEXT_STATE_5) && Message_ShouldAdvance(play)) {
+        func_801477B4(play);
+        this->actionFunc = func_80BFC674;
+    }
+}
+
+void func_80BFC674(EnRz* this, PlayState* play) {
+    EnRz_UpdateSkelAnime(this, play);
+
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        this->actionFunc = func_80BFC608;
+        if (Player_GetMask(play) == PLAYER_MASK_KAMARO) {
+            Message_StartTextbox(play, 0x2925, &this->actor);
+            return;
+        }
+        Message_StartTextbox(play, 0x2924, &this->actor);
+        return;
+    }
+
+    if (EnRz_CanTalk(this, play)) {
+        func_800B8614(&this->actor, play, 120.0f);
+    }
+}
+
+void func_80BFC728(EnRz* this, PlayState* play) {
+    EnRz_UpdateSkelAnime(this, play);
+
+    if ((Message_GetState(&play->msgCtx) == TEXT_STATE_5) && Message_ShouldAdvance(play)) {
+        func_801477B4(play);
+        this->actionFunc = func_80BFC7E0;
+        this->actor.textId++;
+        if (!EN_RZ_GET_PARAM_8000(&this->actor)) {
+            if (this->actor.textId >= 0x2920) {
+                this->actor.textId = 0x291C;
+            }
+        } else if (this->actor.textId >= 0x2924) {
+            this->actor.textId = 0x2920;
+        }
+    }
+}
+
+void func_80BFC7E0(EnRz* this, PlayState* play) {
+    s32 new_var;
+
+    EnRz_UpdateSkelAnime(this, play);
+
+    new_var = this->timer;
+    if (new_var > 0) {
+        this->timer--;
+    } else {
+        this->actionFunc = func_80BFC8F8;
+        func_80BFB9E4(play, this, 2);
+    }
+
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        this->actionFunc = func_80BFC728;
+        func_80BFB9E4(play, this, 0);
+        this->actor.speedXZ = 0.0f;
+        func_80BFBDFC(play);
+    } else if (EnRz_CanTalk(this, play)) {
+        func_800B8614(&this->actor, play, 120.0f);
+    }
+}
+
+void func_80BFC8AC(EnRz* this, PlayState* play) {
+    this->timer = 100;
+    this->actionFunc = func_80BFC7E0;
+    this->actor.speedXZ = 0.0f;
+    func_80BFB9E4(play, this, 0);
+}
+
+void func_80BFC8F8(EnRz* this, PlayState* play) {
+    s32 temp_v0;
+
+    EnRz_UpdateSkelAnime(this, play);
+
+    this->actor.speedXZ = 1.5f;
+    temp_v0 = func_80BFBB44(this);
+    if (temp_v0 != 1) {
+        if (temp_v0 == 2) {
+            func_80BFBA50(this, play);
+            func_80BFC8AC(this, play);
+        }
+    } else {
+        func_80BFC8AC(this, play);
+    }
```

Comment:
```suggestion
    switch(func_80BFBB44(this)) {
        case 1:
            func_80BFC8AC(this, play);
            break;
        case 2:
            func_80BFBA50(this, play);
            func_80BFC8AC(this, play);
            break;
    }
```
Does something like this work? Possibly flipping the cases so 2 is above 1?